### PR TITLE
first cut at clarifying generation of reference identifiers

### PR DESCRIPTION
**Pull request by stpeter**
Fixes https://github.com/richsalz/draft-ietf-uta-rfc6125bis/issues/63

### CODE REVIEW

**Comment by stpeter** [comment]
@martinthomson would you mind also taking a look at this text?

File: draft-ietf-uta-rfc6125bis.md [hunk 1]
```diff
@@ -678,30 +679,37 @@ portion of a username), a hyperlink in a web page that triggers a browser to
...
-information that can be securely parsed out of the inputs, such as parsing
-the FQDN or IP address out of the "host" component or deriving the application service type
-from the scheme of a URI.  Other possibilities include pulling the data from
-a delegated domain that is explicitly established via client or system
-configuration or resolving the data via {{DNSSEC}}.
-These considerations apply only to extraction of the source domain from the
-inputs.  Naturally, if the inputs themselves are invalid or corrupt (e.g., a
-user has clicked a link provided by a malicious entity in a phishing attack),
+This document does not precisely define how reference identifiers are generated,
+and doing so is the responsibility of applications or protocols that use this
```
Comment (richsalz) [inline draft-ietf-uta-rfc6125bis.md:683]: s/and/since/

File: draft-ietf-uta-rfc6125bis.md [hunk 2]
```diff
@@ -678,30 +679,37 @@ portion of a username), a hyperlink in a web page that triggers a browser to
...
+This document does not precisely define how reference identifiers are generated,
+and doing so is the responsibility of applications or protocols that use this 
+document. Because the security of a system that uses this document will depend 
+on how reference identifiers are generated, great care should be taken in this 
+process. For example, a protocol or application could specify that the application 
+service type is obtained through a one-to-one mapping of URI schemes to service 
+types or support only a restricted set of URI schemes. Similarly, it could 
+insist that a domain name or IP address taken as input to the reference 
+identifier must be obtained in a secure context such as a hyperlink embedded 
+in a web page that was delivered over an authenticated and encrypted channel.
```
Comment (richsalz) [inline draft-ietf-uta-rfc6125bis.md:691]: s/encrypted/tamper-proof/ maybe?

**Review by richsalz — APPROVED** [review]
two suggestions, feel free to ignore both.

File: draft-ietf-uta-rfc6125bis.md (hunk 2 above)
Comment (stpeter) [reply draft-ietf-uta-rfc6125bis.md:691]: I was essentially re-using concepts from https://www.w3.org/TR/secure-contexts/ (we could even reference that in a non-normative manner).

Comment (richsalz) [reply draft-ietf-uta-rfc6125bis.md:691]: Yes, please add an informative ref, and maybe put that into the text above if you want.

**Review by martinthomson — APPROVED** [review]
This flows pretty well.  I think that it is a distinct improvement.

I only have one suggestion (in response to Rich's nit).

**Pull request merged**